source/core_mqtt_serializer.c

```diff
@@ -1272,7 +1276,8 @@ static MQTTStatus_t deserializePublish( const MQTTPacketInfo_t * pIncomingPacket
         {
             *pPacketId = UINT16_DECODE( pPacketIdentifierHigh );
 
-            LogDebug( ( "Packet identifier %hu.", *pPacketId ) );
+            LogDebug( ( "Packet identifier %hu.",
+                        ( unsigned short int ) *pPacketId ) );
```

Comment:
Nit: you don't need the `int` here since it's already a `short`, and it would make it consistent with how `unsigned long` is used instead of `unsigned long int`